aws/resource_aws_s3_bucket_notification_test.go

```diff
@@ -364,7 +366,7 @@ resource "aws_sns_topic" "topic" {
 		"Effect": "Allow",
 		"Principal": {"AWS":"*"},
 		"Action": "SNS:Publish",
-		"Resource": "arn:aws:sns:*:*:%s",
+		"Resource": "arn:%s:sns:*:*:%s",
```

Comment:
Rather than requiring this new `partition` string to be passed around as function inputs, which gets messy, I think we should just add a consistently named `aws_partition` data source anywhere its required, e.g. `${data.aws_partition.current.partition}`. This applies to all the testing configurations that have these manually constructed ARNs. 👍

Reply:
Good call--I'll use the `aws_partition` data source where I can. There are a few places where it looks like we'll need to look up partition directly, like in https://github.com/terraform-providers/terraform-provider-aws/pull/4466/files/e80abe873bb68f23cce3ae545e44db2f6687ad57#diff-2f90975e6a0864163009708350852a9aR19, where we're making assertions about a bucket policy and need to build the expected document.